docs/root/intro/arch_overview/service_discovery.rst

```diff
@@ -96,6 +96,14 @@ The Envoy project provides reference gRPC implementations of EDS and
 in both `Java <https://github.com/envoyproxy/java-control-plane>`_
 and `Go <https://github.com/envoyproxy/go-control-plane>`_.
 
+.. _arch_overview_service_discovery_types_custom:
+
+Custom cluster
+^^^^^^^^^^^^^^
+
+Envoy also supports custom cluster discovery mechanism such as `Redis Cluster <https://redis.io/topics/cluster-spec>`_.
```

Comment:
This isn't actually supported yet, so I would make this generic for now, and we can link to a configuration section in the docs when redis cluster actually gets added (I would make yourself a note in this when you add docs for the redis cluster feature).